packages/esm-patient-medications-app/src/order-basket/order-basket-search/order-basket-search-results.component.tsx

```diff
@@ -22,9 +23,21 @@ export default function OrderBasketSearchResults({
 }: OrderBasketSearchResultsProps) {
   const { t } = useTranslation();
   const isTablet = useLayoutType() === 'tablet';
-  const { drugs, isLoading, error } = useDrugSearch(searchTerm);
+  const [currentTerm, setCurrentTerm] = useState(searchTerm);
 
-  if (!searchTerm) {
+  useEffect(() => {
+    const debouncedSetCurrentTerm = debounce((term) => {
+      setCurrentTerm(term);
+    }, 300);
+
+    debouncedSetCurrentTerm(searchTerm);
+
+    return () => debouncedSetCurrentTerm.cancel();
+  }, [searchTerm]);
+
+  const { drugs, isLoading, error } = useDrugSearch(currentTerm);
+
+  if (!currentTerm) {
```

Comment:
Instead of using a new useEffect and adding a new state "currentTerm", I'd suggest why don't you add the debounce when setting the search term?

Q. This brings another issue that the value entered in the search box is appearing after 300ms.
A. Add the debounce to when setting the search term and don't pass the searchTerm to Searchbox as is done [here](https://github.com/openmrs/openmrs-esm-patient-chart/blob/8ed371c29785d439a85be58e73846082b33a112b/packages/esm-patient-medications-app/src/order-basket/order-basket-search/drug-search.component.tsx#L23)